ha/core/controllers/pcs/cluster_status.py

```diff
@@ -0,0 +1,213 @@
+        self._nodes[PcsConstants.COUNT] = len(nodes)
+
+        for a_node in nodes:
+            node_name = a_node.attrib[PcsConstants.NAME]
+            if a_node.attrib[PcsConstants.ONLINE] == PcsConstants.FALSE:
+                self._nodes[PcsConstants.OFFLINE].append(node_name)
+            elif a_node.attrib[PcsConstants.STANDBY_ON_FAIL] == PcsConstants.TRUE or \
+                    a_node.attrib[PcsConstants.MAINTENANCE] == PcsConstants.TRUE or \
+                    a_node.attrib[PcsConstants.PENDING] == PcsConstants.TRUE or \
+                    a_node.attrib[PcsConstants.SHUTDOWN] == PcsConstants.TRUE:
+                self._nodes[PcsConstants.UNHEALTHY].append(node_name)
+            elif a_node.attrib[PcsConstants.ONLINE] == PcsConstants.TRUE and \
+                    a_node.attrib[PcsConstants.STANDBY] == PcsConstants.TRUE:
+                self._nodes[PcsConstants.STANDBY].append(node_name)
+            elif a_node.attrib[PcsConstants.ONLINE] == PcsConstants.TRUE:
+                self._nodes[PcsConstants.ONLINE].append(node_name)
+            else:
+                self._nodes[PcsConstants.UNHEALTHY].append(node_name)
+
+    def _load_services_health(self):
+        """
+            Read xml output and load online and unhealthy services.
+        """
+        resource_path = "./resources/resource"
+        clone_group_resource_path = "./resources/clone/group/resource"
+        group_resource_path = "./resources/clone/group/resource"
+
+        self._services[PcsConstants.ONLINE] = []
+        self._services[PcsConstants.UNHEALTHY] = []
+
+        if self._output is None:
+            return
+
+        resources = self._output.findall(resource_path)
+        resources.extend(self._output.findall(clone_group_resource_path))
+        resources.extend(self._output.findall(group_resource_path))
+
+        for a_resource in resources:
+            resource_name = a_resource.attrib[PcsConstants.ID]
+            if a_resource.attrib[PcsConstants.ACTIVE] == PcsConstants.FALSE or \
+                    a_resource.attrib[PcsConstants.ORPHANED] == PcsConstants.TRUE or \
+                    a_resource.attrib[PcsConstants.BLOCKED] == PcsConstants.TRUE or \
+                    a_resource.attrib[PcsConstants.FAILED] == PcsConstants.TRUE:
+                self._services[PcsConstants.UNHEALTHY].append(resource_name)
+            elif a_resource.attrib[PcsConstants.ROLE] == PcsConstants.STARTED:
+                self._services[PcsConstants.ONLINE].append(resource_name)
+            else:
+                self._services[PcsConstants.UNHEALTHY].append(resource_name)
+
+    def load(self):
+        """
+            Loads pcs status and stores unhealthy elements internally.
+        """
+        self._get_pcs_status()
+        self._load_nodes_health()
+        self._load_services_health()
+
+    def refresh(self):
+        """
+            Refreshes the internal data structures.
+        """
+        self._nodes_configured = 0
```

Comment:
This should be self._nodes_configured = [].
Does not affect status functionality as refresh is not used. It will get corrected in next PR.